src/components/datetime/datetime.ts

```diff
@@ -619,28 +619,14 @@ export class DateTime extends Ion implements AfterContentInit, ControlValueAcces
       }
     }
 
-    // default to assuming this month has 31 days
-    let numDaysInMonth = 31;
-    let selectedMonth: number;
-    if (monthCol) {
-      monthOpt = monthCol.options[monthCol.selectedIndex];
-      if (monthOpt) {
-        // they have a selected month value
-        selectedMonth = monthOpt.value;
-
-        // calculate how many days are in this month
-        numDaysInMonth = daysInMonth(selectedMonth, selectedYear);
-      }
-    }
-
     // create sort values for the min/max datetimes
     let minCompareVal = dateDataSortValue(this._min);
     let maxCompareVal = dateDataSortValue(this._max);
 
     if (monthCol) {
       // enable/disable which months are valid
       // to show within the min/max date range
-      for (i = 0; i < monthCol.options.length; i++) {
+      for (let i = 0; i < monthCol.options.length; i++) {
```

Comment:
Changed.

I'm new to typescript and didn't know about the preference / performance implications of nested `let` constructs.

Note, I originally didn't notice that `i` was defined at the top of the function -- if you'd prefer that, I can change it back to function scope.